src/Sylius/Component/Core/Grid/Filter/ResourceAutocompleteFilter.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Grid\Filter;
+
+use Sylius\Component\Grid\Data\DataSourceInterface;
+use Sylius\Component\Grid\Filtering\FilterInterface;
+
+final class ResourceAutocompleteFilter implements FilterInterface
+{
+    public function apply(DataSourceInterface $dataSource, string $name, $data, array $options): void
+    {
+        if (empty($data)) {
+            return;
+        }
+
+        $expressionBuilder = $dataSource->getExpressionBuilder();
+        $fields = $options['fields'] ?? [$name];
+        $values = str_contains($data, ',') ? explode(',', $data) : [$data];
```

Comment:
```suggestion
        $values = explode(',', $data);
```

Reply:
But in both `$data` has to be a string. What should happen if it's not? 🤔

Reply:
I've resorted to a simple assertion, since filters are bound to respective form types, and the form type in question should always return a string.